pkg/parser/parser.go

```diff
@@ -0,0 +1,21 @@
+package parser
+
+import (
+	"maps"
+
+	promqlparser "github.com/prometheus/prometheus/promql/parser"
+	"github.com/thanos-io/promql-engine/execution/parse"
+)
+
+var functions = buildFunctions()
```

Comment:
Since this is initialized first, will it get the functions from `EnableExperimentalPromQLFunctions`? 
Seems it does as `TestCanSupportHoltWintersFunc` is able to pass.

Reply:
Yes, I think it works because we call EnableExpermimentalPromQLFunctions when creating a new cortex instance and update the functions map in Prometheus. 

The `parser` is only initialize later on in the code.